### PR TITLE
Integrate testing libraries to the project

### PR DESCRIPTION
**Pull request by kei95**
Ticket: https://jestjs.io/docs/configuration#testenvironment-string

Added following libraries to start begin with:
 - jest: https://jestjs.io
 - ts-jest: https://kulshekhar.github.io/ts-jest
 - react-testing-library: https://testing-library.com/docs/react-testing-library/intro

They will come with several config files, but these are nothing more than just environmental setup. Lemme know if you have any questions about them!

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  presets: [
+    "@babel/preset-env",
+    ["@babel/preset-react", { runtime: "automatic" }],
+    "@babel/preset-typescript",
+  ],
+};
```
Comment (kei95) [inline babel.config.js:7]: babel config to let babel know that we'd run jest + typescript for test

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/** @type {import('ts-jest/dist/types').InitialOptionsTsJest} */
+module.exports = {
+  preset: "ts-jest",
+  testEnvironment: "jest-environment-jsdom",
```
Comment (kei95) [inline jest.config.js:4]: https://jestjs.io/docs/configuration#testenvironment-string

**Review by ryonryon — APPROVED** [review]
lgtm🎉

**Comment by kei95** [comment]
> lgtm🎉

Thanks! I'll add a couple more tests on `button` and merge it to master

File: src/components/Button/Button.test.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import Button from "./Button";
+
+test("Passed tittle - it should be rendered with given props", () => {
```
Comment (ryonryon) [inline src/components/Button/Button.test.tsx]: it is just preference but we can add 
```
describe("<Button /> or Button components", () => {
  ...
})
```

Comment (kei95) [reply src/components/Button/Button.test.tsx]: Okay will add it

File: src/components/Button/Button.tsx [hunk 4]
```diff
@@ -1,28 +1,28 @@
...
-  icon,
-  onClick,
-  ...props
-}: Props) {
+export interface Props extends RootBodyProps {
+  title: string;
+  icon?: ReactElement;
+}
+
+export default function Button({ title, icon, ...rootBodyProps }: Props) {
```
Comment (kei95) [inline src/components/Button/Button.tsx:20]: I came across this file to add `data-testid` property for a test but then figured that I could tweak the props of this file a little. I coupled all props that are passed for `Root` for readability. Lemme know what you think, I'm totally fine reverting this change as it's just my preference here😄

**Comment by kei95** [comment]
This PR is now good to go, @ryonryon plz have a quick look one last time and hit the merge button or let me know it's mergeable so that I can do so

**Review by ryonryon — APPROVED** [review]
👍

**Pull request merged**